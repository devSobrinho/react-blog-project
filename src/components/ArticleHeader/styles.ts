import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};
    padding-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    & ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      /* color: ${theme.colors.darkText}; */
      font-size: ${theme.font.sizes.huge};
    }

    @media${theme.media.lteMedium} {
      & ${HeadingStyles} {
        font-size: ${theme.font.sizes.xlarge};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: 2.4rem;
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
