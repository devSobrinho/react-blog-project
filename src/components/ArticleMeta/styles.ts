import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    p {
      font-size: ${theme.font.sizes.small};
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    @media${theme.media.lteMedium} {
      p {
        font-size: ${theme.font.sizes.small};
      }
    }
  `}
`;
