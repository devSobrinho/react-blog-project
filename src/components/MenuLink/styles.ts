import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    color: ${theme.colors.white};
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    font-size: 1.8rem;
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all, 0.3s ease-in-out;

    & :hover {
      color: ${theme.colors.secondary};
      border-right: 0.5rem solid ${theme.colors.secondary};
    }
  `}
`;
