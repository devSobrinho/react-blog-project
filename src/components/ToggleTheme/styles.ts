import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;

  width: 5rem;
  height: 2.5rem;

  font-size: 0;
  line-height: 0;
  overflow: hidden;
  color: transparent;
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background-color: ${theme.colors.secondary};
    }

    &:checked + ${Slider}:before {
      background-color: ${theme.colors.primary};
    }

    &:focus + ${Slider}:before {
      box-shadow: 0 0 0.5rem ${theme.colors.darkGray};
    }
    /* &:focus:checked + ${Slider}:before {
      box-shadow: 0 0 1rem ${theme.colors.secondary};
    } */

    &:checked + ${Slider}:before {
      transform: translateX(calc(100% - 2px));
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    cursor: pointer;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    background-color: ${theme.colors.mediumGray};
    border-radius: 2rem;
    border: 1px solid ${theme.colors.mediumGray};

    transition: all 300ms ease-in-out;

    display: flex;
    align-items: center;

    &::before {
      content: '';
      position: absolute;
      left: 0.4rem;

      height: 2.1rem;
      width: 2.1rem;
      background-color: ${theme.colors.white};
      border-radius: 50%;

      transition: all 300ms ease-in-out;
    }
  `}
`;
