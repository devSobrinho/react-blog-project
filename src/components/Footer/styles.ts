import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
