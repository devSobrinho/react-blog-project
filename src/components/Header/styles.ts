import styled, { css } from 'styled-components';
import { Title as HeaddingStyles } from '../Heading/styles';

export const Content = styled.div`
  ${({ theme }) => css`
    max-width: 48rem;
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      margin-left: 0;

      & > h2 {
        text-align: center;
      }

  `}
`;
export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    padding-bottom: ${theme.spacings.xhuge};
    margin: 0 auto;

    & ${HeaddingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeaddingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }
    }
  `}
`;
