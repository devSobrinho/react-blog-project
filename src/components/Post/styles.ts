import styled, { css } from 'styled-components';
import { Container as HtmlContentStyles } from '../HtmlContent/styles';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};

    & ${HtmlContentStyles} {
      margin-top: ${theme.spacings.xhuge};
    }
  `}
`;
