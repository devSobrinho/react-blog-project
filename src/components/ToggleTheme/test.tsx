import { screen, fireEvent } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { ToggleTheme } from '.';
import { theme } from '../../styles/theme';

describe('<ToggleTheme />', () => {
  beforeEach(() => {
    localStorage.setItem(
      'theme',
      JSON.stringify({ ...theme, name: 'inverted' }),
    );
  });

  afterEach(() => {
    localStorage.removeItem('theme');
  });

  it('should render input without theme', () => {
    localStorage.removeItem('theme');

    renderTheme(<ToggleTheme />);

    const inputCheckbox = screen.getByRole('checkbox', { checked: false });

    expect(inputCheckbox).not.toBeChecked();
  });

  it('should render input with default theme', () => {
    localStorage.removeItem('theme');

    localStorage.setItem(
      'theme',
      JSON.stringify({ ...theme, name: 'default' }),
    );

    renderTheme(<ToggleTheme />);

    const inputCheckbox = screen.getByRole('checkbox', { checked: false });

    expect(inputCheckbox).not.toBeChecked();
  });

  it('should render input checked with inverted theme', () => {
    renderTheme(<ToggleTheme />);
    const inputCheckbox = screen.getByRole('checkbox', { checked: true });

    expect(inputCheckbox).toBeChecked();
  });

  it('should render input and toggle between theme when toggling input', () => {
    const { container } = renderTheme(<ToggleTheme />);
    const inputCheckbox = screen.getByRole('checkbox', { checked: true });

    expect(inputCheckbox).toBeChecked();
    //input checkbox clicked and active attribute checked
    fireEvent.click(inputCheckbox);
    expect(inputCheckbox).not.toBeChecked();

    fireEvent.click(inputCheckbox);
    expect(inputCheckbox).toBeChecked();

    expect(container.firstChild).toMatchSnapshot();
  });
});
