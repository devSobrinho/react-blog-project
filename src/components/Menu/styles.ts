import styled, { css } from 'styled-components';
import { Container as LogoLinkStyles } from '../LogoLink/styles';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as MenuLinkStyles } from '../MenuLink/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

export const Wrapper = styled.menu`
  ${() => css`
    position: fixed;
    top: 0;
    z-index: 1;
  `}
`;

const menuVisibleMobile = () => css`
  padding: 3px;
  margin-left: calc(100vw - 39px - 1.6rem);
`;
const menuVisibleDesk = () => css`
  margin-left: 14rem;
`;

export const MenuOpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    border-radius: 5px;
    border-color: transparent;
    padding: 5px;
    margin-left: 1.6rem;
    margin-top: 1.6rem;
    transition: all 300ms ease-in-out;
    &:hover {
      color: ${theme.colors.secondary};
    }

    ${menuVisible && menuVisibleDesk()}

    @media ${theme.media.lteSmall} {
      ${menuVisible && menuVisibleMobile()}
    }
  `}
`;

export const Container = styled.nav<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    width: ${menuVisible ? '20rem' : '1rem'};
    transition: all 300ms ease-in-out;
    height: 100vh;
    background: ${theme.colors.primary};
    padding: ${menuVisible ? theme.spacings.medium : '0.5rem'};
    overflow-y: auto;
    overflow: auto;
    overflow-wrap: anywhere;

    & ${HeadingStyles} {
      width: auto;
      margin-top: 5rem;
    }

    & ${LogoLinkStyles} img {
      border: 0.5rem solid ${theme.colors.secondary};
      border-radius: 50%;
    }

    .menu-navbar {
      list-style: none;
      padding: 0;
      margin: 0;
      width: auto;
    }

    .navbar-item ${MenuLinkStyles} {
    }

    @media ${theme.media.lteSmall} {
      width: ${menuVisible ? '100vw' : '0px'};
      padding: ${menuVisible ? theme.spacings.medium : '0rem'};

      .menu-navbar {
        margin: 0 20px;
      }
      .navbar-item ${MenuLinkStyles} {
        width: initial;
        border: none;
      }
    }
  `}
`;
