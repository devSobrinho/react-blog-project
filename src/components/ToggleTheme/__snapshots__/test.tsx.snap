// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ToggleTheme /> should render input and toggle between theme when toggling input 1`
] = `
.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  font-size: 0;
  line-height: 0;
  overflow: hidden;
  color: transparent;
}

.c5:checked + .c3 {
  background-color: #dc143c;
}

.c5:checked + .c3:before {
  background-color: #000000;
}

.c5:focus + .c3:before {
  box-shadow: 0 0 0.5rem #AAAAAA;
}

.c5:checked + .c3:before {
  -webkit-transform: translateX(calc(100% - 2px));
  -ms-transform: translateX(calc(100% - 2px));
  transform: translateX(calc(100% - 2px));
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c2:checked + .c3 {
  background-color: #dc143c;
}

.c2:checked + .c3:before {
  background-color: #FFFFFF;
}

.c2:focus + .c3:before {
  box-shadow: 0 0 0.5rem #CCCCCC;
}

.c2:checked + .c3:before {
  -webkit-transform: translateX(calc(100% - 2px));
  -ms-transform: translateX(calc(100% - 2px));
  transform: translateX(calc(100% - 2px));
}

.c4 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #F9F9F9;
  border-radius: 2rem;
  border: 1px solid #F9F9F9;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4::before {
  content: '';
  position: absolute;
  left: 0.4rem;
  height: 2.1rem;
  width: 2.1rem;
  background-color: #000000;
  border-radius: 50%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    Toogle light and dark modes
    <input
      class="c2"
      type="checkbox"
    />
    <span
      class="c3 c4"
    />
  </label>
</div>
`;
